### PR TITLE
Feature/school library people

### PR DESCRIPTION
**Pull request by topeogunleye**
### Added Changes

- Implemented classes and objects in Ruby.
- Implemented encapsulation and inheritance with Ruby.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  private
+
+  def of_age?
+    age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission == true
+  end
```
Comment (Meltrust) [inline person.rb]: - [ ] This method shouldn't be private.  Kindly move this method above the "private" keyword so it becomes public.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Student Class Inheriting From Person Class
+class Student < Person
+  def initialize(classroom)
+    super
+    @classroom = classroom
+  end
```
Comment (Meltrust) [inline student.rb]: - [ ] This method should have the arguments from the parent, and "classroom" added to it.  Super should initialize the parent method with its required arguments like this:

```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
end
```

That way, the class will be properly initialized! 💪

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Teacher Class Inheriting From Person Class
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (Meltrust) [inline teacher.rb]: - [ ] This method also should have the arguments from the parent, and "specialization" added to it.  Super should initialize the parent method with its required arguments like this:

```ruby
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```
That way, the class will be properly initialized! 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @topeogunleye,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice readme ✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 2 above)
Comment (topeogunleye) [reply student.rb]: @Meltrust, Thank you so much for your review :star:

File: teacher.rb (hunk 3 above)
Comment (topeogunleye) [reply teacher.rb]: @Meltrust, Thank you so much for your review :star:

File: person.rb (hunk 1 above)
Comment (topeogunleye) [reply person.rb]: @Meltrust, Thank you so much for your review :star:

**Review by Meltrust — APPROVED** [review]
Hi @topeogunleye,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**